Factories/README.md

```diff
@@ -0,0 +1,91 @@
+
+## Build instructions
+
+```bash
+mkdir build
+cd build
+
+# This step takes a bit of time as the externals will be built.
+cmake ..
+
+make
+```
+
+## How to use
+
+You have the choice to either create histograms or trees. This choice is made by running the correct executable (either `histFactory.exe` to create histograms, or `treeFactory.exe` to create a tree)
+
+### Setup environment
+
+ - Standalone mode: on `ingrid` and `lxplus`, source `setup_env.sh`
+ - In CMSSW mode: a `cmsenv` is enough
+
+### Use condor
+
+See condorExample.py for usage.
+
+### On-the-fly code generation
+
+Steps are the same if you want to generate trees or histograms. Details below assume you want to create histograms. If it's not the case, simply replace `plotter` by `skimmer` in the instructions.
+
+ 1) First, the code for the plotter needs to be generated. For that, you need to run `createPlotter.exe`. This will create a bunch of C++ code responsible for creating all the histograms you specified. The specification of these histograms is still done in a python script, as before.
+
+ 2) Once the code is generated, it needs to be compiled in order to produce an executable. This executable will produce the plots for a given sample. The specification of the samples is still done in JSON, as before.
+
+An helper script is available to create a viable working directory for the plotter (some dependencies are needed if you want to successfully build the code). This script, `createPlotter.sh`, is created automatically in your build directory after you execute the `cmake` command. An example of usage is:
+
+```
+./createPlotter.sh <skeleton ROOT file> <python configuration file> <output directory>
+```
+
+3 arguments are needed:
+ 1) `skeleton ROOT file`: The absolute path of a ROOT file from one of your datasets. It's needed because of code creating the plotting code needs to know the structure of the tree to correctly identify what is a branch in your plot / cut strings. This can be *any* file from *any* dataset, it really does not matter: you just have to ensure that the tree structure is correct.
+
+ 2) `python configuration file`: Path to your python script generating the list of plots.
+
+ 3) `output directory`: The name of the directory where the code of the plotter will be created. This directory must not exist, otherwise the script will exit with an error message.
+
+
+In addition to generate the code, the script will also build it for you. You should now have a working executable named `plotter.exe` inside the `build` directory of the `output directory`.
+
+The last step is to create the plots. You need to run the plotter like this
+
+```
+./plotter.exe -d <JSON file>
+```
+
+where `JSON file` is a path to a JSON file describing your dataset. Multiple datasets per file are supported.
+
+You can find example of valid python file for both histograms and trees, as well as a sample JSON file in the `test` folder.
+
+### Python file format
```

Comment:
Thanks a lot for the updated readme!

If you want you can copy-paste what I had added in #118, here: https://github.com/cp3-llbb/CommonTools/pull/118/files#diff-8bf2efea51f3743202582ff1ec192b46R83